### DISCUSSION TITLE
Can't show argument list with Ctrl+Shift+A

### DISCUSSION THREAD

**Discussion by uaqeel** [Excel-DNA/ExcelDna#585] [General]
Hello,

I have this method that would show arguments list with Ctrl+Shift+A:

[ExcelFunction(Category = "ZeusXL_RTD", Description = "Returns a Contract's current market by RTD")]
        public static object GetMarket(string Contract, string TickType) {}

I can see the arguments list + the description in the Function Wizard but the keyboard shortcut (Ctrl+Shift+A) no longer works.

I've been away from this addin + Exceldna for a long time, so not sure if something has changed or if this is an Excel problem?

Thank you for your help!

**Comment by govert** [comment]
It works fine on my machine.
Is it only a problem for the UDF, or built-in functions too?

**Reply by uaqeel** [reply]
For built-in functions like SUM(), the moment I put the opening bracket the argument list pops up. For UDFs (mine and Bloomberg's), it doesn't seem to work at all.

I'm running Office 365 (Excel Version 2302) on Windows 11.

Thanks as always Govert.

**Comment by govert** [comment]
Ctrl+Shift+A just fills in the list of parameters of a function into the formula editor. That part still works for me.
![CtrlShiftA](https://user-images.githubusercontent.com/414659/228819537-f672416c-1e0e-4427-acdc-b50c5d3933f6.gif)

So what you describe is not what Ctrl+Shift+A does - you are talking about the IntelliSense pop-up that displays argument information.
Excel does not show IntelliSense for UDF functions, but I have made a custom add-in that adds this feature - see https://github.com/Excel-DNA/IntelliSense/

You can add this support to your add-in by installing the "ExcelDna"IntelliSense" NuGet package, then adding some code to your project:
```c#
    using ExcelDna.Integration;
    using ExcelDna.IntelliSense;
    public class AddIn : IExcelAddIn
    {
        public void AutoOpen()
        {
            // Versions before v1.1.0 required only a call to Register() in the AutoOpen().
            // The name was changed (and made obsolete) to highlight the pair of function calls now required.
            IntelliSenseServer.Install();
        }

        public void AutoClose()
        {
            IntelliSenseServer.Uninstall();
        }
    }
```

Then for a function like this
```c#
        [Description("is a useful function that calculates something.")]
        public static object DoCalc([Description("is the first value")] double val1,
                                    [Description("is the other value")] double val2) 
            => val1 + val2;
```

you get

![PopUp](https://user-images.githubusercontent.com/414659/228821939-32b588cd-7ba0-4266-a3d5-47a9a9d55cd8.gif)

**Comment by uaqeel** [comment]
Thanks Govert. Even this doesn’t happen for me right now for UDFs:

 

 <https://user-images.githubusercontent.com/414659/228819537-f672416c-1e0e-4427-acdc-b50c5d3933f6.gif> 

 

I will add the Intellisense package as you suggest.

 

One additional question, my code (which as I mentioned is many years old now) used ExcelDna.Contrib in order to use the ExcelRange class. I think at some point this was moved to ExcelDna proper, but when I updated my packages just now it seems this class is no longer part of ExcelDna.Integration? Here’s the code that’s getting tripped up:

 

        [ExcelCommand(MenuName = "&Zeus", MenuText = "&Tidy Array", ShortCut = "^+T", Name = "TidyArray")]

        public static void TidyArray()

        {

            ExcelRange selection = new ExcelRange(XlCall.Excel(XlCall.xlfSelection, new object[0]) as ExcelReference);

            if (selection.HasFormula)

            {

                string formula = selection.Formula;

                //string newFormula = Regex.Replace(formula, "([A-Z]{1,2}[0-9]+)", "!$1");

                string newFormula = Regex.Replace(formula, @"(\${0,1}[A-Z]{1,2}\${0,1}[0-9]+)", "!$1");

                newFormula = Regex.Replace(newFormula, "!!", "!");

 

                object value = XlCall.Excel(XlCall.xlfEvaluate, newFormula);

                if (value is ExcelDna.Integration.ExcelError || value is ExcelDna.Integration.ExcelReference ||

                    (value.GetType().IsArray && ((object[,])value)[0, 0] is ExcelDna.Integration.ExcelError))

                    value = XlCall.Excel(XlCall.xlfEvaluate, formula);

 

                if (value is object[,])

                {

                    object[,] values = value as object[,];

                    int nRows = values.GetLength(0), nCols = values.GetLength(1);

 

                    ExcelReference newRef = new ExcelReference(selection.AsRef.RowFirst, selection.AsRef.RowFirst + nRows - 1,

                                                               selection.AsRef.ColumnFirst, selection.AsRef.ColumnFirst + nCols - 1);

                    XlCall.Excel(XlCall.xlcSelect, newRef);

                }

            }

        }

 

Obviously now Excel has its own auto-expand functionality for array functions so this isn’t strictly required for me, but I’d like to get it back into working shape if possible!

 

As ever, thank you!
